### PR TITLE
[HttpClient][DI] Add an option to use the MockClient in functional tests

### PR DESCRIPTION
**Pull request by GaryPEGEOT**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | TODO

Let's say you want to mock HTTP responses in your test or dev environment:

First, you create an invokable class responsible of generating the response:
```php
namespace App\Tests;

use Symfony\Contracts\HttpClient\ResponseInterface;

class MyAwesomeCallback
{
    public function __invoke(string $method, string $url, array $options = []): ResponseInterface
    {
        // load a fixture file or generate dynamic data
    }
}
```

Then configure it:
```yaml

# config/services_test.yaml
services:
   # ...
   App\Tests\MyAwesomeCallback: ~

# config/packages/test/framework.yaml
framework:
    http_client:
        mock_response_factory: App\Tests\MyAwesomeCallback
```

The HttpClient will now be using MockHttpClient in your functional test:
```php
$client = static::createClient();

// No live HTTP connection made
$client->request('GET', '/path-with-http-call');
```

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Very related to #35651 and #35677

**Comment by fabpot** [comment]
@nicolas-grekas Can you have a look at this one as it would unblock some other PRs as well?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -1450,6 +1450,9 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode)
                                 ->end()
                             ->end()
                         ->end()
+                        ->scalarNode('mock_response_factory')
+                            ->info('ID of the service used to generate mock responses. Must be invokable.')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php:1454]: The id of the service that should generate mock responses. It should be either an invokable or an iterable.

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php [hunk 2]
```diff
@@ -1567,6 +1568,21 @@ public function testMailerWithSpecificMessageBus(): void
         $this->assertEquals(new Reference('app.another_bus'), $container->getDefinition('mailer.mailer')->getArgument(1));
     }
 
+    public function testHttpClientMockResponseFactory(): void
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php:1571]: ~`: void`~

**Review by nicolas-grekas — APPROVED** [review]
LGTM, I just have minor notes.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (GaryPEGEOT) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Fixed

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @GaryPEGEOT.

**Pull request merged**